lib/python/milc.py

```diff
@@ -178,15 +178,25 @@ def __init__(self, parent, *args, **kwargs):
 
     def __getitem__(self, key):
         """Returns a config value, pulling from the `user` section as a fallback.
+        This is called when the attribute is accessed either via the get method or through [ ] index.
         """
-        if key in self._config:
+        if key in self._config and self._config[key]:
```

Comment:
I'm not sure this is the right thing to do here. There are a lot of values that evaluate false that I think we should return here, like `False` and `""`. However, it's been a bit since I've looked at this code so I might be missing context. Would something like this work here?

```suggestion
        if self._config.get(key) is not None:
```

Reply:
I try to remember why I did it that way, but I'm also in favour of comparing to `None` rather that `False`.
However, I believe the correct way would be `if key in self._config and self._config.get(key) is not None:`.

What do you think?

Reply:
It's a choice between performance and readability. You can avoid the exception handling in `self._config.get(key)` by checking `key in self._config` first but I find it's also easier to misread the actual check when you're reading code. Since this isn't performance critical code I tend to prefer the simpler form, but both are OK here.